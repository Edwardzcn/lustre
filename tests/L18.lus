node main(b : bool) returns (x : int)
let
  x = if b then x else 1;
tel
