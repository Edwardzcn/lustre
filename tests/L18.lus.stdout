PhaseToCore
===========

node (b : bool)
  returns (x)
  assumes ()
  shows ()
let
  rec
    x : int
      = ITE(b, x, 1)
tel

--- Step 1 ---
b : bool  = False
x = 1
--- Step 2 ---
b : bool  = False
x = 1
--- Step 3 ---
b : bool  = (EOF)
